force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,151 @@
+public with sharing class OpportunityHelper {
+    public List<Opportunity> newRecords;
+    public List<Opportunity> oldRecords;
+    public Map<Id,Opportunity> newRecordsMap;
+    public Map<Id,Opportunity> oldRecordsMap;
+
+    public OpportunityHelper() {
+        this.newRecords = Trigger.new;
+        this.oldRecords = Trigger.old;
+        this.newRecordsMap = (Map<Id, Opportunity>) Trigger.newMap;
+        this.oldRecordsMap = (Map<Id, Opportunity>) Trigger.oldMap;
+    }
+
+    ///OpportunityTrigger.trigger implementation  ===== Opportunity Part 1
+
+    public static void oppAmountValidation (List<Opportunity> newRecords) {
+        for(Opportunity opp : newRecords){
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+    public static void closedWonOppDeletion(List<Opportunity> oldRecords){
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN (SELECT AccountId FROM Opportunity WHERE Id IN :oldRecords)]);
+        for(Opportunity opp : oldRecords){
+            if(opp.StageName == 'Closed Won'){
+                if(accounts.get(opp.AccountId).Industry == 'Banking'){
+                    opp.addError('Cannot delete a closed won opportunity for a banking account');
+                }
+            }
+        }
+    }
```

Comment:
Another way to do this would be to only query for the accounts for Opps that are closed won and filter the query for banking. 

```
  private void validateDeletedRecords(){
        Set<Id> accountIds = new Set<Id>();
        for(Opportunity opportunity : (List<Opportunity>)Trigger.old){
            if (opportunity.StageName == 'Closed Won') {
                accountIds.add(opportunity.AccountId);
            }
        }
        Map<Id, Account> bankingAccountsById  = new Map<Id, Account>([SELECT Id, Industry
        FROM Account
        WHERE Id IN :accountIds
        AND Industry = 'Banking']);

        for(Opportunity opportunity : (List<Opportunity>)Trigger.old){
            if(bankingAccountsById.containsKey(opportunity.AccountId) ){
                opportunity.addError('Cannot delete closed opportunity for a banking account that is won');
            }
        }
    }
  ```

Reply:
Thanks @bdJohnson72. Your code will help me avoid nested if loops.